pandas/tests/indexes/test_datetimelike.py

```diff
@@ -721,6 +721,17 @@ def test_fillna_datetime64(self):
                            dtype=object)
             self.assert_index_equal(idx.fillna('x'), exp)
 
+    def test_contains(self):
+        #GH13572
+        dates = ['2015-01-03', '2015-01-01', '2015-01-04', '2015-01-05', '2015-01-02']
+        monotonic = pd.to_datetime(sorted(dates))
+        non_monotonic = pd.to_datetime(['2015-01-03', '2015-01-01', '2015-01-04', '2015-01-05', '2015-01-02'])
```

Comment:
Thx to confirm. Pls leave other nat-likes (``pd.NaT, None, float('nan'), np.nan``) ATM because ``PeriodIndex`` is not fully supported them yet (I'll cover in #12759, but PR is appreciated).